src/components/LogView/LogLineText.js

```diff
@@ -16,11 +16,52 @@ type Props = {
   port: ?string,
   text: string,
   highlightText: string[],
-  handleDoubleClick: ()=> void,
+  handleDoubleClick: () => void,
+  prettyPrint: boolean,
+}
+
+export function findJSONObjectsInLine(text: string) {
+  var startIndex = 0;
+  var numBraces = 0;
+  var chunks: string[] = [];
+  for (var i = 0; i < text.length; i++) {
+    if (text[i] === '{') {
+      if (numBraces === 0 && i !== 0) {
+        const lineBreak = (startIndex === 0) ? '' : '\n';
+        chunks.push(lineBreak + text.substring(startIndex, i));
```

Comment:
We need a way to not add line breaks if there are curly braces, but not actually json. I added a line to a log with the text `i am not json { 123 } something mismatched {{{ a {{ }}}}}}}}}}` which added some line breaks, but shouldn't have been changed

Reply:
Fixed- my newline placement was messed up, it is both cleaner and much easier to understand now.